### ISSUE TITLE
Build failure: Common/types.h is missing

### ISSUE THREAD

**Issue by SpaghettiToastBook** [arukibree/PrimeWorldEditor#15]
Excerpt from `ninja`'s output:
```
[2/244] Building CXX object src/Core/CMakeFiles/pwe_core.dir/CMayaSpline.cpp.o
FAILED: src/Core/CMakeFiles/pwe_core.dir/CMayaSpline.cpp.o 
/usr/bin/c++  -DGLEW_STATIC -DUNICODE -DUSE_LZOKAY -DWITH_CODEGEN -I../src/Core/.. -I../src/Core -I../src/Core/OpenGL -I../externals/LibCommon/Source -I../externals/LibCommon/Externals/CodeGen/include -isystem ../.dew/prefix-release/include -isystem ../.dew/prefix-release/include/lzokay -O3 -DNDEBUG   -Wno-multichar -Wno-undefined-var-template -std=gnu++17 -MD -MT src/Core/CMakeFiles/pwe_core.dir/CMayaSpline.cpp.o -MF src/Core/CMakeFiles/pwe_core.dir/CMayaSpline.cpp.o.d -o src/Core/CMakeFiles/pwe_core.dir/CMayaSpline.cpp.o -c ../src/Core/CMayaSpline.cpp
In file included from ../src/Core/CMayaSpline.cpp:1:
../src/Core/CMayaSpline.h:4:10: fatal error: Common/types.h: No such file or directory
 #include <Common/types.h>
          ^~~~~~~~~~~~~~~~
```

**Comment by SpaghettiToastBook** [comment]
I'm building on Arch Linux. I remembered to update the submodules.

**Comment by arukibree** [comment]
Oops, sorry. That should be Common/BasicTypes.h. Will fix soon.

**Comment by SpaghettiToastBook** [comment]
Thanks!

**Comment by SpaghettiToastBook** [comment]
I changed that and tried building again and got this:
```
In file included from ../src/Core/CMayaSpline.cpp:1:
../src/Core/CMayaSpline.h:5:10: fatal error: Math/CVector2f.h: No such file or directory
 #include <Math/CVector2f.h>
          ^~~~~~~~~~~~~~~~~~
```
I'm guessing something similar happened here?

**Comment by arukibree** [comment]
Should be <Common/Math/CVector2f.h>

I was pretty sure I tested that this file compiled before committing it but I guess evidently I didn't. Weird... not sure what happened.

**Comment by SpaghettiToastBook** [comment]
One more:
```
../src/Core/CMayaSpline.cpp:2:10: fatal error: Math/MathUtil.h: No such file or directory
 #include <Math/MathUtil.h>
          ^~~~~~~~~~~~~~~~~
```
I tried changing this to `Common/Math/MathUtil.h` but got a bunch of errors about `u32` and `u8` not naming a type.

**Comment by arukibree** [comment]
Should be fixed in cd7937f73c9019c62c20c1bbee61f69ba5de6e9d. Apologies for the delay. I haven't been able to 100% confirm that the project compiles without issue yet since I am working through some additional issues that came up when reviewing the recent build system changes. Let me know if you're able to build now.

**Comment by SpaghettiToastBook** [comment]
I just built it successfully :)
